doc/user/layouts/shortcodes/create-source/syntax-details.html

```diff
@@ -21,6 +22,26 @@
 
 ## Details
 
+### Materialized source details
+
+By default, sources in Materialize do not retain data from the external
+components to which they connect. Instead, [materialized
+views](../../overview/api-components/#materialized-views) and their attendant
+[indexes](../../overview/api-components/#indexes) retain and manage data. The
+presence of these indexes is necessary to make the views `SELECT`-able; the
+absence of indexes are why sources are not `SELECT`-able by default.
+
+However, you can choose to "materialize" a source, which creates an index that
+stores _all_ of the data the source receives. This has the benefit of making the
+source `SELECT`-able, but that must be weighed against the cost of retaining all
+of the source's data for the source's lifetime.
```

Comment:
This isn't exactly right. We maintain the source's data compacted, so at any time the retained footprint is roughly proportional to the *current* size of the source, modulo a bit of compaction that may not yet have happened.

Would it help to work on some text about "Materialization, Generally" as a concept that we can throw folks at? This same stuff will come up with materialized views as well (and Indexes, as the common concept backing these two shorthands).